hris_publisher/hris.py

```diff
@@ -0,0 +1,187 @@
+        :param entry: a single json entry from the hris.json
+        :return: bool (truthy)
+        """
+        try:
+            hris_schema_json = os.path.join(os.path.abspath(os.path.dirname(__file__)), 'schema.json')
+
+            with open(hris_schema_json, 'r') as schema_data:
+                hris_schema_json = json.load(schema_data)
+
+            jsonschema_validate(entry, hris_schema_json)
+            return True
+        except ValidationError:
+            return False
+
+    def to_groups(self, entry):
+        """
+
+        :param entry: from the hris.json
+        :return: a list of groups to reintegrate to cis beginning with hris_
+        """
+        g = Groups(entry=entry)
+        return g.all
+
+    def _locate_s3_bucket(self):
+        """Fine the S3 bucket in the account using bucket tags."""
+        if self.boto_session is None:
+            self.boto_session = self._connect_boto_session()
+
+        client = self.boto_session.client('s3', region_name='us-west-2')
+        s3 = self.boto_session.resource('s3')
+        for this_bucket in client.list_buckets().get('Buckets'):
+            bucket_name = this_bucket.get('Name')
+            bucket_tagging = s3.BucketTagging(bucket_name)
+
+            try:
+                tag_set = bucket_tagging.tag_set
+                if self._is_tag_match(tag_set):
+                    self.bucket_name = bucket_name
+                    return bucket_name
+            except ClientError:
+                # Boto3 throws client error when a bucket has no tags.
+                continue
+
+    def _is_tag_match(self, tag_set):
+        for tag in tag_set:
+            if tag.get('Value') == 'hris_publisher':
+                return True
+        return False
+
+    def _connect_boto_session(self):
+        return boto3.session.Session()
+
+    def _get_object_from_bucket(self):
+        """Get the byte stream of the hris.json file."""
+        if self.boto_session is None:
+            self.boto_session = self._connect_boto_session()
+
+        client = self.boto_session.client('s3', region_name='us-west-2')
+        response = client.get_object(
+            Bucket=self.bucket_name,
+            Key=self.file_name
+        )
+        return response.get('Body')
+
+    def _read_object(self, streaming_body):
+        """Return the content of the file object byte stream we retrieved."""
+
+        return json.loads(streaming_body.read())
+
+
+class Groups(object):
+    """Return a list of groups for a given profile based on rules."""
+    def __init__(self, entry):
+        self.hris_entry = entry
+        self.hris_grouplist = []
+
+    @property
+    def all(self):
+        """Run all rules and return the grouplist from the constructor."""
+        self.cost_center_rule()
+        self.cost_center_hierarchy()
+        self.management_level_rule()
+        self.manager_name_rule()
+        self.manager_status_rule()
+        self.egencia_country_rule()
+
```

Comment:
add:
`self.is_staff_rule()`

which checks if `CurrentlyActive` is 1 and add to group accordingly, which means user is staff
if `CurrentlyActive` is 0, do not integrate any group data at all for `hris_` instead, and and change the `user.active` attribute to False instead

Note: we'll probably want an additional safety check that looks if all users that were listed as active in the workday file, are the ones in the vault, and otherwise adjust + warn accordingly. This is especially important when LDAP stops getting workday data.

Reply:
Interesting fact about this file... there are no users in it that have currently active set to 0

Reply:
I'm highly suspecting this means we need to go through the list of users and mark them inactive if they are not in the hris/workday list. cc @jabba